scripts/subselect.py

```diff
@@ -0,0 +1,229 @@
+bl_grp.add_argument("--max-ew-length", type=float, help="Maximum EW baseline length to select, metres", default=None)
+
+io_grp = ap.add_argument_group("I/O")
+io_grp.add_argument("--clobber", action="store_true", help="Overwrite output file if it exists.")
+
+check_grp = ap.add_argument_group("Checking Arguments")
+check_grp.add_argument("--check", action="store_true", help="Check output file for consistency with input file.")
+check_grp.add_argument("--check-acceptability", action="store_true", help="Check output file for consistency with input file.")
+check_grp.add_argument("--check-uvw-strict", action="store_true", help="Check whether UVW's are strictly consistent with antpos.")
+check_grp.add_argument("--check-autos", action="store_true", help="Check that autos are real")
+check_grp.add_argument("--fix-autos", action="store_true", help="Fix autos if they are not real")
+
+args = parse_args(ap)
+kw = filter_kwargs(vars(args))
+
+def select(
+    infile: str, outfile: str, 
+    freq_min=None, freq_max=None, freq_spws=None, time_min=None, time_max=None, 
+    time_idxs=None, lst_min=None, lst_max=None, lst_in_hours=False, pols=None, 
+    antennas=None, calfile=None, bls=None, only_autos=False, only_cross=False, 
+    min_bl_length=None, max_bl_length=None, min_ew_length=None, max_ew_length=None,
+    clobber=False,  check=False, check_acceptability=False, check_uvw_strict=False,
+    check_autos=False, fix_autos=False
+):
+    if not os.path.exists(infile):
+        raise FileNotFoundError(f"File {ap} does not exist.")
+    if infile != outfile and os.path.exists(outfile) and not clobber:
+        raise FileExistsError(f"File {outfile} exists and clobber is False.")
+    if infile == outfile:
+        clobber= True
+
+    hd = UVData()
+    logger.info(f"Reading metadata from file {infile}")
+    hd.read(infile, read_data=False)
+    hd.use_future_array_shapes()
+
+    # Get frequencies
+    if freq_min is not None or freq_max is not None or freq_spws is not None:
+        logger.info("Getting frequencies to read.")
+        freqs = hd.freq_array.copy()
+        if freq_min is not None:
+            freqs = freqs[freqs >= freq_min]
+        if freq_max is not None:
+            freqs = freqs[freqs <= freq_max]
+        if freq_spws is not None:
+            freq_spws = [tuple(map(int, spw.split("~"))) for spw in freq_spws.split(",")]
+            freqs = freqs[np.concatenate([np.arange(*spw) for spw in freq_spws])]
+    else:
+        freqs = None
+
+    # Get times
+    if time_min is not None or time_max is not None or time_idxs is not None:
+        logger.info("Getting times to read.")
+        times = np.unique(hd.time_array)
+        time_bools = np.ones_like(times, dtype=bool)
+
+        if time_idxs is not None:
+            time_idxs = [tuple(map(int, idx.split("~"))) for idx in time_idxs.split(",")]
+            for idx in time_idxs:
+                time_bools[idx[0]:idx[1]] = False
```

Comment:
We want to keep times between idx[0] and idx[1] right? wouldn't setting this to False deselect them?

It might be easier (to wrap a reader's head around) if the boolean array defaults to False, then selected times are set to True. It would also match the logic style from the freqs block above then